hr_employee_lastnames/models/hr_employee.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons.hr_employee_firstname.models.hr_employee import UPDATE_PARTNER_FIELDS
+
+UPDATE_PARTNER_FIELDS += ['lastname2']
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    lastname2 = fields.Char("Second last name")
+
+    @api.model
+    def _get_name_lastnames(self, lastname, firstname, lastname2=None):
+        return self.env['res.partner']._get_computed_name(
+            lastname, firstname, lastname2)
+
+    def _build_name_in_create(self, vals):
```

Comment:
You are overwriting this method without super...

Is it possible using super and post-process with extra field all methods?

Reply:
@moylop260 

No, because the original method processes 'vals' and I don't need to do that, I require the data as it originally comes to set name or firstname, lastname and lastname2 according to the case.

Reply:
```suggestion
    def _build_name_in_create(self, vals):
        values = vals.copy()
        res = super(HrEmployee, self)._build_name_in_create(vals)
        # your logic with values or vals as you consider
```

Reply:
done